src/Splat.Autofac/AutofacDependencyResolver.cs

```diff
@@ -83,26 +83,31 @@ public void SetLifetimeScope(ILifetimeScope lifetimeScope)
         }
 
         /// <inheritdoc />
-        public virtual IEnumerable<object> GetServices(Type serviceType, string? contract = null)
+        public virtual IEnumerable<object> GetServices(Type? serviceType, string? contract = null)
         {
+            if (serviceType is null)
+            {
+                throw new ArgumentNullException(nameof(serviceType));
+            }
+
             lock (_lockObject)
             {
                 try
                 {
                     var enumerableType = typeof(IEnumerable<>).MakeGenericType(serviceType);
                     var instance = Resolve(enumerableType, contract);
 
-                    if (instance is null)
+                    if (instance is not null)
                     {
-                        return Array.Empty<object>();
+                        return new object[] { instance };
                     }
-
-                    return ((IEnumerable)instance).Cast<object>();
                 }
                 catch (DependencyResolutionException)
                 {
-                    return Array.Empty<object>();
+                    // no op
                 }
+
+                return Enumerable.Empty<object>();
```

Comment:
Array.Empty<T> is better over Enumerable.Empty<T> so much the case that they changed it to be an analog in .NET Core, but for Net Framework better to keep Array.Empty<T>

Reply:
Array.Empty where possible